packages/csvviewer/src/widget.ts

```diff
@@ -81,12 +81,17 @@ class CSVViewer extends Widget {
     layout.addWidget(this._grid);
 
     context.pathChanged.connect(this._onPathChanged, this);
-    // Throttle the rendering rate of the widget.
-    this._monitor = new ActivityMonitor({
-      signal: context.model.contentChanged,
-      timeout: RENDER_TIMEOUT
+
+    this._context.ready.then(() => {
+      this._updateGrid();
+      this._ready.resolve(undefined);
```

Comment:
Here the previous `_updateGrid()` seems to make sure that the widget is ready. But later in other widgets, it isn't obvious. Will add comments below...